src/Cake.Svn/Delete/SvnDeleter.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using Cake.Core;
+using Cake.Svn.Internal.Extensions;
+
+namespace Cake.Svn.Delete
+{
+    /// <summary>
+    /// Class for deleting files or directories from Subversion directory trees.
+    /// </summary>
+    public sealed class SvnDeleter : SvnTool<SvnDeleteSettings>
+    {
+        private readonly ICakeEnvironment _environment;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SvnDeleter"/> class.
+        /// </summary>
+        /// <param name="environment">The Cake environment.</param>
+        /// <param name="clientFactoryMethod">Method to use to initialize a Subversion client.</param>
+        public SvnDeleter(ICakeEnvironment environment, Func<ISvnClient> clientFactoryMethod)
```

Comment:
`environment` is currently not used and can be removed. But see below regarding handling of absolute / relative paths for which you might use it.